src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -15,11 +15,33 @@
   </Target>
 
   <PropertyGroup>
-    <PrepareForRunDependsOn Condition="'$(IsTestProject)'=='true'">$(PrepareForRunDependsOn);CopyTestToTestDirectory</PrepareForRunDependsOn>
+    <PrepareForRunDependsOn Condition="'$(IsTestProject)'=='true'">$(PrepareForRunDependsOn);CopyTestToTestDirectory;CreateTestNuGetPackage</PrepareForRunDependsOn>
     <TestArchitecture Condition="'$(TestArchitecture)' == ''">x64</TestArchitecture>
     <TestNugetRuntimeId Condition="'$(TestNugetRuntimeId)' == ''">win7-$(TestArchitecture)</TestNugetRuntimeId>
   </PropertyGroup>
 
+  <Target Name="CreateTestNuGetPackage"
+          DependsOnTargets="CopyTestToTestDirectory"
+          Condition="'$(GenerateTestPackages)'=='true'">
+    <PropertyGroup>
+      <TestPackageContent>@(_TestCopyLocalByFileNameWithoutDuplicates->'%(SourcePath)')</TestPackageContent>
+      <TestPackageBuild Condition="'$(TestPackageBuild)'==''">$(BuildNumberMajor)</TestPackageBuild>
+      <TestPackageBuild Condition="'$(TestPackageBuild)'==''">000000</TestPackageBuild>
+      <TestPackageVersion Condition="'$(TestPackageVersion)'==''">1.0.0-prerelease-$(TestPackageBuild)</TestPackageVersion>
+    </PropertyGroup>
+    <MSBuild Projects="$(MSBuildThisFileDirectory)testpackage.nuproj"
+      Properties="
+      IntermediateOutputPath=$(IntermediateOutputPath);
+      OutputPath=$(OutputPath);
+      NuProjPropsPath=$(NuProjDir)NuProj.props;
```

Comment:
Hi Wes, I saw you found where this is set in https://github.com/dotnet/corefx/pull/3762#discussion-diff-41768313R45.

I put the definition of NuProjDir there following the pattern of PackagesDir being set in the corefx repo and consumed in the buildtools repo.  Since NuProjDir depends on PackagesDir and NuProjVersion both of which we know in buildtools, I could move the definition of NuProjDir here, and agree that would be better.  I'll update both pull requests with that change.